modules/gatk4/baserecalibratorspark/main.nf

```diff
@@ -0,0 +1,53 @@
+process GATK4_BASERECALIBRATOR_SPARK {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::gatk4=4.2.3.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gatk4:4.2.3.0--hdfd78af_0' :
+        'broadinstitute/gatk:4.2.3.0' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index), path(intervals)
+    path  fasta
+    path  fai
+    path  dict
+    path  known_sites
+    path  known_sites_tbi
+
+    output:
+    tuple val(meta), path("*.table"), emit: table
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def intervals_command = intervals ? "-L ${intervals}" : ""
+    def sites_command = known_sites.collect{"--known-sites ${it}"}.join(' ')
+
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[GATK BaseRecalibrator] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" BaseRecalibratorSpark \\
+        -R $fasta \\
+        -I $input \\
+        $sites_command \\
+        $intervals_command \\
+        --tmp-dir . \\
```

Comment:
I am torn whether this should go in `modules.config` it's quite important, but technically not a file. Eitherway should be the same for all gatk modules